From 87e18e5c00f36e10031b4b878c27a828d59280cf Mon Sep 17 00:00:00 2001
From: Aleksander Jan Bajkowski <olek2@wp.pl>
Date: Fri, 14 May 2021 21:25:08 +0200
Subject: [PATCH 3/4] MIPS: lantiq: dma: increase burst length for ethernet

Increase the burst length for Ethernet. This improves Ethernet
performance by 25%.

The NAT benchmark results on xRX200:
* 1W: 332 Mb/s
* 2W: 330 Mb/s
* 4W: 432 Mb/s    372 Mb/s
* 8W: 520 Mb/s    389 Mb/s

Tested on xRX200 and xRX330.

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>
---
 arch/mips/lantiq/xway/dma.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/mips/lantiq/xway/dma.c b/arch/mips/lantiq/xway/dma.c
index 88f25d4656fd..293a277a65d4 100644
--- a/arch/mips/lantiq/xway/dma.c
+++ b/arch/mips/lantiq/xway/dma.c
@@ -190,8 +190,9 @@ ltq_dma_init_port(int p)
 		 * Tell the DMA engine to swap the endianness of data frames and
 		 * drop packets if the channel arbitration fails.
 		 */
-		ltq_dma_w32_mask(0, DMA_ETOP_ENDIANNESS | DMA_PDEN,
-			LTQ_DMA_PCTRL);
+		ltq_dma_w32_mask(0x3c, ((DMA_PCTRL_8W_BURST << DMA_TX_BURST_SHIFT) |
+			(DMA_PCTRL_8W_BURST << DMA_RX_BURST_SHIFT) | DMA_ETOP_ENDIANNESS |
+			DMA_PDEN), LTQ_DMA_PCTRL);
 		break;
 
 	case DMA_PORT_DEU:
-- 
2.30.2

